src/sagemaker_training/process.py

```diff
@@ -177,20 +183,22 @@ def check_error(cmd, error_class, processes_per_host, cwd=None, capture_error=Tr
         subprocess.Popen: The process for the given command.
 
     Raises:
-        error_class: If there is an exception raised when creating the process.
+        ExecuteUserScriptError: If there is an exception raised when creating the process.
     """
 
     if capture_error:
         return_code, output, process = create(
             cmd,
-            error_class,
+            error_classes,
             processes_per_host,
             env=os.environ,
             cwd=cwd or environment.code_dir,
             capture_error=True,
             **kwargs,
         )
-        stderr = output[1]
+        stderr = " ".join(output)
```

Comment:
output is a list of size 2. The `output[0]` has stdout and `output[1]` has stderr.  Why are we combining both here? The `stdout` might contain normal log messages as well. Could you please clarify?